packages/report-server/src/__tests__/reportBuilder/query/QueryBuilder.test.ts

```diff
@@ -82,23 +82,33 @@ describe('QueryBuilder', () => {
   const outputQuery = (input: Partial<FetchReportQuery> = {}) => ({
     hierarchy: HIERARCHY,
     organisationUnitCodes: ['TO'],
-    period: getPeriodString([2017, 1], [2020, 12]),
+    period: getPeriodString([2017, 1], [2020, 12], 15),
     startDate: '2017-01-01',
-    endDate: '2020-12-31',
+    endDate: '2020-12-15',
     ...input,
   });
 
   const getPeriodString = (
     [startYear, startMonth]: [number, number],
     [endYear, endMonth]: [number, number],
+    endDay?: number,
+    startDay?: number,
```

Comment:
Personally I'd prefer to use this syntax, which allows passing an array of numbers of length 1,2, or 3
```suggestion
    [startYear, startMonth, startDay]: [number, number?, number?],
    [endYear, endMonth, endDay]: [number, number?, number?],
```

However it has one drawback which seems like a bug in Typescript to me tbh. It treats the type of `startMonth` and `startDay` as `number` when  really they should be `number | undefined` imo... It's fine cos we can just manually check for them in the function but still...